src/Symfony/Component/Scheduler/Scheduler.php

```diff
@@ -62,9 +66,17 @@ public function run(array $options = []): void
 
             $ran = false;
             foreach ($this->generators as $generator) {
-                foreach ($generator->getMessages() as $message) {
-                    $this->handlers[$message::class]($message);
+                foreach ($generator->getMessages() as $context => $message) {
+                    $preRunEvent = new PreRunEvent($generator->schedule(), $context, $message);
+                    $this->dispatcher?->dispatch($preRunEvent);
+
+                    if (!$preRunEvent->hasBeenCancelled()) {
+                        $this->handlers[$message::class]($message);
```

Comment:
missing `continue` here?

Reply:
I'll push an update because it can't depend on the event: the event is optional